app/javascript/packs/building_map.js

```diff
@@ -1,70 +1,86 @@
-import { displayRoute, setupMap } from './leafletMap.js';
+import { displayRoute, setupMap, pins } from './leafletMap.js';
 
 let currentLocation;
 const YOUR_LOCATION_MAGIC_STRING = "Your location" // This will be changed when the page supports multiple languages
+const PIN_1_MAGIC_STRING = "Pin 1"
+const PIN_2_MAGIC_STRING = "Pin 2"
 
 setupMap();
 
-const start_input_field = $("#start_input")[0];
-start_input_field.addEventListener("change", () => {
+const startInputField = $("#start_input")[0];
+startInputField.addEventListener("change", () => {
     request_location();
     if (validate_place_input("start_input", "startOptions")) {
-        start_input_field.setCustomValidity("");
+        startInputField.setCustomValidity("");
     } else {
-        start_input_field.setCustomValidity(
+        startInputField.setCustomValidity(
             "Please select a valid starting place."
         );
     }
 });
-start_input_field.dispatchEvent(new Event("change"));
+startInputField.dispatchEvent(new Event("change"));
 
-const dest_input_field = $("#dest_input")[0];
-dest_input_field.addEventListener("change", () => {
+const destInputField = $("#dest_input")[0];
+destInputField.addEventListener("change", () => {
     if (validate_place_input("dest_input", "destOptions")) {
-        dest_input_field.setCustomValidity("");
+        destInputField.setCustomValidity("");
     } else {
-        dest_input_field.setCustomValidity(
+        destInputField.setCustomValidity(
             "Please select a valid destination place."
         );
     }
 });
-dest_input_field.dispatchEvent(new Event("change"));
+destInputField.dispatchEvent(new Event("change"));
 
 $("#navigation_form")[0]
     .addEventListener("submit", (event) => {
         event.preventDefault();
-        if (start_input_field.value === YOUR_LOCATION_MAGIC_STRING) {
-            start_input_field.value = currentLocation;
+        for(let inputField of [startInputField, destInputField]){
+            switch(inputField.value){
+                case YOUR_LOCATION_MAGIC_STRING:
+                    inputField.value = currentLocation;
+                    break;
+                case PIN_1_MAGIC_STRING:
+                    inputField.value = pinCoordinatesString(pins[0]);
+                    break;
+                case PIN_2_MAGIC_STRING:
+                    inputField.value = pinCoordinatesString(pins[1]);
+                    break;
+            }
         }
-        const start = start_input_field.value;
-        const dest = dest_input_field.value;
+        const start = startInputField.value;
+        const dest = destInputField.value;
         displayRoute(start, dest);
     });
 
+function pinCoordinatesString(pin){
+    return String(pin.getLatLng().lat.toFixed(7)) + "," + String(pin.getLatLng().lng.toFixed(7))
+}
+
 function validate_place_input(inputId, optionsId) {
```

Comment:
```suggestion
function validatePlaceInput(inputId, optionsId) {
```
You should be consistent with this (even though it's not "your" code)

Reply:
PS: don't forget to change this, wherever it's used as well